docs/using-oci-flexvolume-driver.md

```diff
@@ -0,0 +1,140 @@
+
+## OCI Policies
+
+You must ensure the user (or group) associated with the OCI credentials provided has the following level of access. See [policies][7] for more information.
+
+TODO: The below is now out of date as we require mounting in a kubeconfig.
+
+```
+"Allow group id GROUP to read vnic-attachments in compartment id COMPARTMENT",
+"Allow group id GROUP to read vnics in compartment id COMPARTMENT"
+"Allow group id GROUP to read instances in compartment id COMPARTMENT"
+"Allow group id GROUP to read subnets in compartment id COMPARTMENT"
+"Allow group id GROUP to use volumes in compartment id COMPARTMENT"
+"Allow group id GROUP to use instances in compartment id COMPARTMENT"
+"Allow group id GROUP to manage volume-attachments in compartment id COMPARTMENT"
+```
+
+### Kubernetes DaemonSet Installer
+
+The recommended way to install the driver is through the DaemonSet installer
+mechanism. This will create two DaemonSets, one specifically for master nodes,
+allowing configuration via a Kubernetes Secret, and one for worker nodes.
+
+```bash
+$ kubectl apply -f https://github.com/oracle/oci-flexvolume-driver/releases/download/${flexvolume_driver_version}/rbac.yaml
+$ kubectl apply -f https://github.com/oracle/oci-flexvolume-driver/releases/download/${flexvolume_driver_version}/oci-flexvolume-driver.yaml
+```
+
+You'll still need to add the config file as a Kubernetes Secret.
+
+Once the Secret is set and the daemonsets deployed, the configuration file will be placed onto the master nodes.
+
+## Assumptions
+
+- If a Flexvolume is specified for a Pod, it will only work with a single
+  replica. (or if there is more than one replica for a Pod, they will all have
+  to run on the same Kubernetes Node). This is because a volume can only be
+  attached to one instance at any one time. Note: This is in common with both
+  the Amazon and Google persistent volume implementations, which also have the
+  same constraint.
+
+- If nodes in the cluster span availability domain you must make sure your Pods are scheduled
+  in the correct availability domain. This can be achieved using the label selectors with the zone/region.
+
+  Using the [oci-volume-provisioner][3] makes this much easier.
+
+- For all nodes in the cluster, the instance display name in the OCI API must
+  match with the instance hostname, start with the vnic hostnamelabel or match the public IP.
+  This relies on the requirement that the nodename must be resolvable.
+
+## Limitations
+
+Due to [kubernetes/kubernetes#44737][6] ("Flex volumes which implement
+`getvolumename` API are getting unmounted during run time") we cannot implement
+`getvolumename`. From the issue:
+
+> Detach call uses volume name, so the plugin detach has to work with PV Name
+
+This means that the Persistent Volume (PV) name in the `pod.yml` _must_ be
```

Comment:
I am not sure what voice we are using across the docs. Here is this paragraph with the pronouns removed. See what you think:

This means that the Persistent Volume (PV) name in the `pod.yml` _must_ be
the last part of the block volume OCID ('.' separated). Otherwise, there would
be no way of determining which volume to detach from which worker node. Even
if the state was stored at the time of volume attachment; PV names would have
to be unique across the cluster which is an unreasonable constraint.